### DISCUSSION TITLE
Poor performance of `bcrypt` for password verification

### DISCUSSION THREAD

**Discussion by muhammadyusuf-kurbonov** [feathersjs/feathers#3158] [General]
I noticed that password verification runs for more than 1 sec. Only verification. Also, I have made a test of `bcrypt`. The same 1 second. So will you use another library or do something with performance? Thank you for your attention

**Comment by pdfowler** [comment]
We ran into a similar issue and found that the [default hash function in authentication-local](https://github.com/feathersjs-ecosystem/authentication-local/blob/9b17f48b32e30bd5961886b87efa6ebb93c5947f/lib/utils/hash.js#L7) was taking _a very long time_. We came up with the following customHasher:

```
export async function customHasher(password: string) {
  const salt = await bcrypt.genSalt(BCRYPT_WORK_FACTOR_BASE);
  return await bcrypt.hash(password, salt);
}
```

This was a problem we saw on signup, not verification; but I suspect the two issues are related
